### ISSUE TITLE
[BUILD] embedded fmtlib link errors on Windows

### ISSUE THREAD

**Issue by yann-lty** [OpenImageIO/oiio#2276]
### Problem
OIIO version: 2.0.8
The embedded version of fmtlib contains definitions of isnan/_isnan functions that clash with Windows' ones. It does not prevent OIIO from building, but can cause link errors when using OIIO as a library (probably due to Windows' <math.h> also being included at some point):
```
error C2084: function 'fmt::v5::internal::dummy_int fmt::v5::internal::_isnan(...)' already has a body (compiling source file ...) ...\fmt\format.h(278): note: see previous definition of '_isnan'.
```
This was reported on fmtlib (https://github.com/fmtlib/fmt/issues/1104) and a fix has been merged in master since. No official release including this fix has been made yet, but using the last master snapshot (https://github.com/fmtlib/fmt/tree/2711cb1672543cc67b5f9a2b1ec900d0b333f93c ATM) solves the problem.
Would it be possible to upgrade the embedded version of fmtlib despite the lack of an official release?

**Comment by lgritz** [comment]
Yes, I'll look into upgrading fmt. Has the fix made it into any tagged release of fmt, or do I need to lock to a certain individual commit for now?

**Comment by lgritz** [comment]
Patch proposed in #2280. Any chance you can test on your end? (It's a patch on master, but if it solves the problem, I think it's safe to backport to 2.0.)

**Comment by yann-lty** [comment]
Thanks a lot! I've just tested it and it fixed the link errors due to fmtlib. 
But using master, I stumbled upon another error in fmath.h, again related to isnan/_isnan. After further investigation, it turned out that this error was actually due to another external library with a toxic ```#define isnan _isnan``` that was the root of all problems, and not an issue with conflicting Windows's headers.
I re-tested building against OIIO 2.0.8 and everything is actually fine. 
Sorry for the trouble, I guess fmtlib update can be delayed until an official release if it feels safer.
